src/logics/ArithLogic.h

```diff
@@ -8,16 +8,6 @@
 #include <numeric>
 
 namespace opensmt {
-class LANonLinearException : public std::runtime_error {
-public:
-    LANonLinearException(char const * reason_) : runtime_error(reason_) {
-        msg = "Term " + std::string(reason_) + " is non-linear";
-    }
-    virtual char const * what() const noexcept override { return msg.c_str(); }
-
-private:
-    std::string msg;
-};
```

Comment:
I think we need to keep it for now, but we need to do the check elsewhere, when we are trying to process the terms inside LASolver.

Reply:
Done